File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = -1

    conv_s = []
    
    for i in s:
        if len(conv_s) == 0 or conv_s[-1] != i:
            conv_s.append(i)
        else:
            conv_s.pop()
    
    return 1 if len(conv_s) == 0 else 0

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;

class Solution
{   
    static Stack<Character> stack = new Stack<>();
    
    public int solution(String s)
    {
        
        for(char c : s.toCharArray()){
            if(stack.isEmpty()){ //스택 비어있으면 그냥 push
                stack.push(c);
            }else if(stack.peek().equals(c)){ //스택 맨 위의 값이 현재값과 같으면 pop
                stack.pop();
            }else{
                stack.push(c); //스택 맨 위의 값이 현재값과 다르면 push
            }
        }
            
        return stack.isEmpty() ? 1 : 0;
    }
}